### ISSUE TITLE
Avoid unbound variable false positive when iterating over non-empty tuple

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#1438]
Refer to this issue: https://github.com/microsoft/pylance-release/issues/496

```python
def best_unit(dt: float) -> Tuple[float, str]:
    units = (
        (1.0, "sec"),
        (1e-3, "msec"),
        (1e-6, "usec"),
        (1e-9, "nsec"),
    )
    for scale, unit in units:
        if dt >= scale:
            break
    return scale, unit  # "unit" is possibly unbound Pylance(reportUnboundVariable)
```

**Comment by erictraut** [comment]
I've decided to change the code flow logic for for loops to eliminate this false positive. The tradeoff is that some actual errors may be missed. For example:

```python
for i in []:
   pass
print(i)
```
In this case, `i` is unbound and generates a runtime exception. The previous logic would have reported this problem. The updated technique is silent in this case.

**Comment by erictraut** [comment]
This is addressed in Pyright 1.1.107, which I just published. It will also be included in the next release of Pylance.
